netket/graph/lattice.py

```diff
@@ -246,6 +315,114 @@ def atoms_coord(self):
+
+            perms.append(tuple(perm))
+        return tuple(perms)
+
+    def rotation_perm(self, period, axes=[0, 1]):
+        perm = []
+        axes = list(axes)
+        angle = 2 * pi / period
+        rot_mat = _np.array(
+            [[_np.cos(angle), -_np.sin(angle)], [_np.sin(angle), _np.cos(angle)]]
+        )
+
+        rot_coords = self._coords.copy()
+        rot_coords[:, axes] = _np.matmul(rot_coords[:, axes], rot_mat)
+
+        for hash_coord in rot_coords:
+            hash_coord = _np.matmul(hash_coord, self._inv_dims) % 1
+            hash_coord = _np.around(hash_coord * 10 ** tol_digits).astype(int) % (
+                10 ** tol_digits
+            )
+            hash_coord = HashableArray(hash_coord)
+            if hash_coord in self._hash_positions:
+                perm.append(self._hash_positions[hash_coord])
+            else:
+                raise ValueError(
+                    "Rotation with the specified period and axes does not map lattice to itself"
+                )
+
+        return tuple(perm)
+
+    def reflection_perm(self, axis=0):
+        perm = []
+        ref_coords = self._coords.copy()
+        ref_coords[:, axis] = -1 * ref_coords[:, axis]
+
+        for hash_coord in ref_coords:
+            hash_coord = _np.matmul(hash_coord, self._inv_dims) % 1
+            hash_coord = _np.around(hash_coord * 10 ** tol_digits).astype(int) % (
+                10 ** tol_digits
+            )
+            hash_coord = HashableArray(hash_coord)
+            if hash_coord in self._hash_positions:
+                perm.append(self._hash_positions[hash_coord])
+            else:
+                raise ValueError(
+                    "Reflection about specified axis does not map lattice to itself"
+                )
+
+        return tuple(perm)
+
+    def planar_rotations(self, period, axes=[0, 1]) -> SymmGroup:
+        """
+        Returns SymmGroup corresponding to rotations about specfied axes with specified period
+
+        Arguments:
+            period: Period of the rotations
+            axes: Axes that define the plane of the rotation
+        """
+
+        perm = self.rotation_perm(period, axes)
+        rotations = [PlanarRotation(perm, n) for n in range(1, period)]
+
+        return SymmGroup([Identity()] + rotations, graph=self)
+
+    def basis_translations(self) -> SymmGroup:
+        """
+        Returns SymmGroup corresponding to translations by basis vectors
+        """
+
+        translations = product(*[range(i) for i in self.extent])
+        next(translations)
+
+        perms = self.translation_perm()
+        translations = [Translation(perms, i) for i in translations]
+
+        return SymmGroup([Identity()] + translations, graph=self)
+
+    def reflections(self, axis=0) -> SymmGroup:
+        """
+        Returns SymmGroup corresponding to reflection about axis
+        args:
+          axis: Generated reflections about specified axis
+        """
```

Comment:
`axis: int = 0`

Dot at the end of the sentence.

Newline between the docstring and the arguments, otherwise the website breaks.
`args:` should be `Args:`, otherwise the website wont work